### ISSUE TITLE
Inline const/let variables

### ISSUE THREAD

**Issue by rentalhost** [swc-project/swc#7765]
### Describe the feature

Hello, I noticed that `const` or `let` variables are not being _inlined_, even when it seems possible. Is there any reason for this?

**For example, the following code:**

```ts
const a = b();
c(a);
```

This code is not being inlined, and it could be just `c(b())`. 

Same will happen with:

```ts
const a = await b();
await c(a);

// same as: await c(await b())
```

https://play.swc.rs/?version=1.3.74&code=H4sIAAAAAAAAA0vOzysuUUhUsFVI0tC05uVK1kjUtAYAekA23RUAAAA%3D&config=H4sIAAAAAAAAA32VwZLbIAyG73mKjM85bHPY6fQBeuszMASEQ2qQB0Q2mZ28e4Vjp85G7CmOPgnQjyQ%2BN9ttd8qm%2B7X95E%2F%2BQ0nH7DCFh4mNCbShlaGaSiQfgI2dLoRBkzfdjG%2FT7213X3HUKUNaL5evkfSlhoIJOpvkR%2Bp2Cz3lipweMjytQzr1QFNU3r%2Ft93NENyBmWCJmW%2FDRu%2Bt6T4NhTJDzcxa8ZAkQqZopFditUcIPwX5AHIA1ahOls%2FKRoIckeBkcBj1mUGct48CC%2BYzSDhUWAqvGhKPIo%2FXkMbIUr9SCtsqgBQH5BIb8GaQw3ovDYubsGtTCofT9dMlfMJz1UDQJgXCZ7oMP%2B8qO6DMpVyQN7qwhwB3KynqnElBJwn4n9LER9ReAsx90zlEHSZ7JwXEhNdl3gT46LlW6vmIuaynDCD3Lqbx3gqZVFUjkpe0S2GKgimraVNYgewsKnOMSkeiHJ3OUtqTrCOgEwBerpfPf7arVejOuXdCmvzlBEqtqduBRdWzCfA0HlJpnCQY6om1zvgHCJk08Ey5jE5dogcsBrORRsmSvfUmKUA3TaHypB24HXlD1Ax7EmTByYU9Xt19m92MOBx37qeNX85JwHOAMw%2FO4ndBro4ge7lvKBz7X8l7a%2B6uTh5%2BClcXjkfnj7enZeDxDmzmpLqAtq4Sm8rw%2FJ%2B%2Fdf6fl5ViE6nz%2BswRW2%2Bb2D5T0NKI1BwAA

**Comment by kdy1** [comment]
Try it on terser playground. This is intentional and please search before filing an issue

**Comment by rentalhost** [comment]
Sorry, you are correct that Terser works the same way, but I couldn't find any information regarding the "why not". Earlier I tried to search for something like "inline const" and now I did a broader search, including Terser's repository, but unfortunately I still couldn't find it.

The terms I looked up were: _inline inlining function call const let var variable_.

Maybe I'm missing something, or I couldn't identify the issue by the title. In fact, it seems like a pretty simple feature not to exist today, so no doubt there's a good reason for it, I'm just curious now why.
